app/exposureHistory.ts

```diff
@@ -17,23 +17,43 @@ export interface NoKnown {
 
 export type ExposureDatum = Possible | NoKnown;
 
-export type ExposureHistory = ExposureDatum[];
-
-const HISTORY_LENGTH = 21;
+export type ExposureInfo = Record<Posix, ExposureDatum>;
 
-export const blankHistory = (): NoKnown[] => {
-  const now = Date.now();
-  const daysAgo = [...Array(HISTORY_LENGTH)].map((_v, idx: number) => {
-    return HISTORY_LENGTH - 1 - idx;
-  });
+export type ExposureHistory = ExposureDatum[];
 
-  return daysAgo.map(
-    (daysAgo: number): NoKnown => {
-      const date = dayjs(now).subtract(daysAgo, 'day').startOf('day').valueOf();
+export const toExposureHistory: (
+  exposureInfo: ExposureInfo,
+  calendary: Posix[],
+) => ExposureHistory = (exposureInfo, calendar) => {
+  return calendar.map((date: Posix) => {
+    if (exposureInfo[date]) {
+      return exposureInfo[date];
+    } else {
       return {
         kind: 'NoKnown',
         date,
       };
+    }
+  });
+};
+
+export const calendarDays = (today: Posix, totalDays: number): Posix[] => {
```

Comment:
this one might want to live in another file as the concept is somewhat independent from the exposure history, though not sure if itll ever be used elsewhere so i think its fine to leave in here for now.